src/XMakeTasks/UnitTests/ResolveAssemblyReference_Tests.cs

```diff
@@ -179,6 +179,19 @@ public void Cleanup()
             Environment.SetEnvironmentVariable("MSBUILDDISABLEASSEMBLYFOLDERSEXCACHE", null);
         }
 
+        protected static readonly string s_rootPathPrefix = NativeMethodsShared.IsWindows ? "C:\\" : Path.VolumeSeparatorChar.ToString();
```

Comment:
One problem with `Environment.GetLogicalDrives()[0]` is that if the first logical drive happens to be a floppy drive, e.g. `A:\`, _and_ that path is checked with e.g. `File.Exists()`, the `File.Exists()` will be *slow*.

If that's a problem, this can instead use:

    protected static readonly string s_rootPathPrefix = System.IO.DriveInfo.GetDrives ()
        .Where (d => d.DriveType == DriveType.Fixed && d.IsReady)
        .Select (d => d.RootDirectory.FullName);